tensordict/tensordict.py

```diff
@@ -1811,13 +1811,12 @@ def as_tensor(self):
         and will raise an exception in all other cases.
 
         """
-        try:
-            return self._fast_apply(lambda x: x.as_tensor())
-        except AttributeError as err:
-            raise AttributeError(
-                f"{self.__class__.__name__} does not have an 'as_tensor' method "
-                f"because at least one of its tensors does not support this method."
-            ) from err
+        def as_tensor(x):
+            try:
+                return x.as_tensor()
+            except AttributeError:
+                return x
+        return self._fast_apply(as_tensor)
```

Comment:
So in this version, no `AttributeError` will be thrown ?

Reply:
The idea is that we want to deprecate this method progressively as MemmapTensor will eventually be a regular tensor thanks to @mikaylagawarecki's work!
Since we call as_tensor here and there and the backend will change, the only way to make this non bc breaking is to bypass errors in as_tensor

Reply:
Thank you for the clarification!